### PR TITLE
DEMO ONLY: implement scopes on interfaces

### PR DESCRIPTION
**Pull request by temaEmelyan**
This is just a demo, do not intend to merge this

### CODE REVIEW

File: test/src/test/resources/fixtures/chained transforms/scopes-on-types-implementing-an-interface-1.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+name: scopes on types implementing an interface 1
+enabled: true
+overallSchema:
+  service: |
+    directive @toBeDeleted on FIELD_DEFINITION | OBJECT
```
Comment (temaEmelyan) [inline test/src/test/resources/fixtures/chained transforms/scopes-on-types-implementing-an-interface-1.yml:5]: `@toBeDeleted` - represents a failed scope (or grant) check in the gateway

File: lib/src/main/java/graphql/nadel/engine/transform/result/NadelResultTransformer.kt [hunk 2]
```diff
@@ -35,15 +35,15 @@ internal class NadelResultTransformer(private val executionBlueprint: NadelOvera
         service: Service,
         result: ServiceExecutionResult,
     ): ServiceExecutionResult {
-        val nodes = JsonNodes(result.data, executionContext.hints)
```
Comment (temaEmelyan) [inline lib/src/main/java/graphql/nadel/engine/transform/result/NadelResultTransformer.kt:38]: just cleanup, ignore

File: test/src/test/resources/fixtures/chained transforms/scopes-on-types-implementing-an-interface-4.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  }
+variables: { }
+serviceCalls:
+    - serviceName: service
+      request:
+        query: |
+          {
+            issue {
+              uuid_typename: __typename
+              ... on ThirdPartyIssue {
```
Comment (temaEmelyan) [inline test/src/test/resources/fixtures/chained transforms/scopes-on-types-implementing-an-interface-4.yml:64]: other types failed scope check, proceed with only querying `ThirdPartyIssue`

**Comment by gnawf** [comment]
Old

**Pull request closed without merging**